### ISSUE TITLE
Regression with documentbuilderfactory-disallow-doctype-decl-missing

### ISSUE THREAD

**Issue by coheigea** [returntocorp/sgrep-rules#2547]
There is a regression with the rule documentbuilderfactory-disallow-doctype-decl-missing due to the fix in:

    update DocumentBuilderFactory rule to use taint and fix FPs

With this latest rule it causes a finding to occur with code that looks like:
```
DocumentBuilderFactory factory = XmlUtils.getSecureDocumentBuilderFactory();
documentBuilder = factory.newDocumentBuilder();
```
Here we are setting the correct features in a method in a separate class. We are never calling DocumentBuilderFactory.newInstance in the class that the error is getting flagged on, but it is still flagging the issue. IMO if the DocumentBuilderFactory is not being instantiated in the class in question it should not flag a finding on newDocumentBuilder.

**Comment by 0xDC0DE** [comment]
Thanks for the report @coheigea ! Should be fixed with the next release of this rule. I think half of the testcode for this rule are your contributions at this point, thanks a lot!

**Comment by coheigea** [comment]
Thanks to you @0xDC0DE  for the fix.
